app/src/main/java/com/arctouch/codechallenge/ui/detail/DetailsViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.arctouch.codechallenge.ui.detail
+
+import android.arch.lifecycle.ViewModel
+import android.util.Log
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.Logger
+import io.reactivex.Observable
+
+class DetailsViewModel : ViewModel() {
+    private var TAG = Logger.tag
+
+    var movieId: Long = 0
+
+    fun requestMovie(): Observable<Movie>? {
```

Comment:
Why was this method declared as returning `Observable<Movie>?` if it always returns an instance?

Reply:
Yes, an instance of an Observable Movie type. Sorry but I think I did not understand your questioning, could you explain me better?

Reply:
Sorry, I meant: the signature of the method states it might return `null` (it has a `?` after `Observable<Movie>`). However, it always returns an instance of `Observable<Movie>`. So why is the `?` there?

Reply:
Ok, now I understand. So, I always like to set it this way, for server and local bank requests. So I already check if any item in the answer is null the "?" already treats this.